### PR TITLE
:triangular_flag_on_post: 添加北京科技大学研究生院router

### PR DESCRIPTION
**Pull request by DA1Y1**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/ustb/yjsy/news/all
/ustb/yjsy/news/cres
/ustb/yjsy/news/seee
/ustb/yjsy/news/metall
/ustb/yjsy/news/mse
/ustb/yjsy/news/me
/ustb/yjsy/news/saee
/ustb/yjsy/news/scce
/ustb/yjsy/news/shuli
/ustb/yjsy/news/huasheng
/ustb/yjsy/news/sem
/ustb/yjsy/news/wenfa
/ustb/yjsy/news/marx
/ustb/yjsy/news/sfs
/ustb/yjsy/news/ncms
/ustb/yjsy/news/skl
/ustb/yjsy/news/iet
/ustb/yjsy/news/cicst
/ustb/yjsy/news/slam
/ustb/yjsy/news/adma
/ustb/yjsy/news/ihmm
/ustb/yjsy/news/sd
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [x] New Route
- [ ] Documentation
  - [x] CN
  - [ ] EN
- [ ] 全文获取 fulltext
  - [ ] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] 日期和时间 date and time
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

### CODE REVIEW

**Review by TonyRL — CHANGES_REQUESTED** [review]
v2 script standard #8236

File: assets/radar-rules.js [hunk 1]
```diff
@@ -1288,4 +1288,15 @@
...
+        _name: '北京科技大学',
+        gs: [
+            {
+                title: '研究生院',
+                docs: 'https://docs.rsshub.app/university.html#bei-jing-ke-ji-da-xue',
+                source: '/:type',
+                target: '/ustb/news/:type',
+            },
+        ],
+    },
```
Comment (TonyRL) [inline assets/radar-rules.js:1301]: Unnecessary addition.
This `radar-rules.js` is here for backwards compatibility.
Rules in /v2/ROUTE/radar.js from all routes will be aggregated into a new file. See https://github.com/DIYgod/RSSHub/issues/8269

File: lib/v2/ustb/maintainer.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    '/ustb/news/marx': ['DA1Y1'],
+    '/ustb/news/sfs': ['DA1Y1'],
+    '/ustb/news/ncms': ['DA1Y1'],
+    '/ustb/news/skl': ['DA1Y1'],
+    '/ustb/news/iet': ['DA1Y1'],
+    '/ustb/news/cicst': ['DA1Y1'],
+    '/ustb/news/slam': ['DA1Y1'],
+    '/ustb/news/adma': ['DA1Y1'],
+    '/ustb/news/ihmm': ['DA1Y1'],
+    '/ustb/news/sd': ['DA1Y1'],
```
Comment (TonyRL) [inline lib/v2/ustb/maintainer.js:23]: Please sort the routes in alphabetical order.

File: lib/v2/ustb/yjsy/news.js [hunk 3]
```diff
@@ -0,0 +1,453 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const iconv = require('iconv-lite');
+// const logger = require('../../../utils/logger');
```
Comment (TonyRL) [inline lib/v2/ustb/yjsy/news.js:4]:
```suggestion
// const logger = require('@/utils/logger');
```

File: lib/v2/ustb/yjsy/news.js [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+                    item = $(item);
+                    // logger.info("=====");
+                    // logger.info("item:" + item);
+                    // time
+                    let time = item.find($(struct[type].timeSelector.list)).text();
+                    if (time.includes('[')) {
+                        time = time.slice(1, 11);
+                    }
+                    const date = new Date(Date.parse(time));
+                    const bj_date = date.getTime() / 1000 + 8 * 60 * 60;
```
Comment (TonyRL) [inline lib/v2/ustb/yjsy/news.js:427]: Please use parseDate()
docs: https://docs.rsshub.app/joinus/pub-date.html

File: lib/v2/ustb/yjsy/news.js [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+                    if (title === '') {
+                        title = item.find($(struct[type].titleSelector.list)).text();
+                    }
+                    // logger.info("title:" + title);
+                    // logger.info("=====");
+                    return {
+                        title,
+                        description: title,
+                        link,
+                        pubDate: new Date(parseInt(bj_date) * 1000),
```
Comment (TonyRL) [inline lib/v2/ustb/yjsy/news.js:448]: Please use parseDate()
docs: https://docs.rsshub.app/joinus/pub-date.html

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/ustb/yjsy/news.js [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+                .map((index, item) => {
+                    item = $(item);
+                    // logger.info("=====");
+                    // logger.info("item:" + item);
+                    // time
+                    let time = item.find($(struct[type].timeSelector.list)).text();
+                    if (time.includes('[')) {
+                        time = time.slice(1, 11);
+                    }
+                    const date = parseDate(Date.parse(time));
```
Comment (TonyRL) [inline lib/v2/ustb/yjsy/news.js:427]:
```suggestion
                    const date = parseDate(time);
```

**Review by TonyRL — CHANGES_REQUESTED** [review]

**Comment by TonyRL** [comment]
`/ustb/news/saee` has some null items

**Comment by DA1Y1** [comment]
update: /ustb/news/* -> /ustb/yjsy/news/*

File: lib/v2/ustb/radar.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'ustb.edu.cn': {
+        _name: '北京科技大学',
+        gs: [
+            {
+                title: '研究生院',
+                docs: 'https://docs.rsshub.app/university.html#bei-jing-ke-ji-da-xue',
+                source: '/:type',
+                target: '/ustb/news/:type',
```
Comment (TonyRL) [inline lib/v2/ustb/radar.js:9]: `target` also needs an update

File: lib/v2/ustb/router.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+module.exports = function (router) {
+    router.get('/yjsy/news/:type?', require('./yjsy/news'));
```
Comment (TonyRL) [inline lib/v2/ustb/router.js:2]:
```suggestion
    router.get('/yjsy/news/:type', require('./yjsy/news'));
```
It's not an optional parameter according to your docs.

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/ustb/router.js (hunk 8 above)
Comment (DA1Y1) [reply lib/v2/ustb/router.js:2]: thank u

File: docs/university.md [hunk 9]
```diff
@@ -180,6 +180,18 @@ pageClass: routes
...
+
+| 北京科技大学研究生院 | 土木与资源工程学院 | 能源与环境工程学院 | 冶金与生态工程学院 | 材料科学与工程学院 | 机械工程学院 | 自动化学院 | 计算机与通信工程学院 | 数理学院 | 化学与生物工程学院 | 经济管理学院 | 文法学院 | 马克思主义学院 | 外国语学院 | 国家材料服役安全科学中心 | 新金属材料国家重点实验室 | 工程技术研究院 | 钢铁共性技术协同创新中心 | 钢铁冶金新技术国家重点实验室 | 新材料技术研究院 | 科技史与文化遗产研究院 | 顺德研究生院 |
+| -------------------- | ------------------ | ------------------ | ------------------ | ------------------ | ------------ | ---------- | -------------------- | -------- | ------------------ | ------------ | -------- | -------------- | ---------- | ------------------------ | ------------------------ | -------------- | ------------------------ | ---------------------------- | ---------------- | ---------------------- | ------------ |
+| all                  | cres               | seee               | metall             | mse                | me           | saee       | scce                 | shuli    | huasheng           | sem          | wenfa    | marx           | sfs        | ncms                     | skl                      | iet            | cicst                    | slam                         | adma             | ihmm                   | sd           |
+
+</Route>
+
 ## 北京科技大学天津学院
 
 ### 北京科技大学天津学院
```
Comment (TonyRL) [inline docs/university.md:197]:
```suggestion
### 天津学院
```

File: lib/v2/ustb/yjsy/news.js [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
+                list: 'li p',
+            },
+            titleSelector: {
+                list: 'li a',
+            },
+            linkSelector: {
+                list: 'li a',
+            },
+            url: 'http://slam.ustb.edu.cn/slam//smvc/ListBs-articleList/SNM2015072600000000000011',
+            link: 'http://slam.ustb.edu.cn/slam//',
```
Comment (TonyRL) [inline lib/v2/ustb/yjsy/news.js:330]: `//` or `/` ?

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/ustb/yjsy/news.js (hunk 10 above)
Comment (DA1Y1) [reply lib/v2/ustb/yjsy/news.js:330]: eg: http://slam.ustb.edu.cn/slam//smvc/ArticleBs/notice/SNA2021070509400178095000

Comment (DA1Y1) [reply lib/v2/ustb/yjsy/news.js:330]: 至于 北京科技大学天津学院 应该是独立于 北京科技大学

**Review by TonyRL — CHANGES_REQUESTED** [review]
`/ustb/yjsy/news/saee` and `/ustb/yjsy/news/iet` are still failing.

**Comment by DA1Y1** [comment]
> `/ustb/yjsy/news/saee` and `/ustb/yjsy/news/iet` are still failing.

however i can get the correct answer
image
image

File: lib/v2/ustb/yjsy/news.js (hunk 10 above)
Comment (DA1Y1) [reply lib/v2/ustb/yjsy/news.js:330]: I enter any announcement from the official website⬇️
http://slam.ustb.edu.cn/slam//smvc/ListBs-articleList/SNM2015072600000000000011
they are both double slash

**Review by TonyRL — APPROVED** [review]

**Pull request merged**